src/store/index.js

```diff
@@ -55,8 +66,44 @@ export default new Vuex.Store({
       const task = tasksArr.find((item) => item.taskId === taskId);
       task.taskState = !task.taskState;
     },
+    // resetStore(state) {
+    //   // for (const key in defaultStore) {
+    //   //   if (defaultStore.hasOwnProperty(key)) {
+    //   //     state[key] = defaultStore[key];
+    //   //   }
+    //   // }
+    //   Object.keys(defaultStore).forEach((key) => {
+    //     state[key] = defaultStore[key];
+    //   });
+    // },
   },
   actions: {
+    async signUserUp({ commit }, { email, password }) {
+      try {
+        const user = await firebase.auth()
+          .createUserWithEmailAndPassword(email, password);
+        const newUser = { id: user.user.uid };
+        commit('setUser', newUser);
+      } catch (err) {
+        alert(err.message);
+      }
```

Comment:
Will be better to move request error handling to component to save store cleaner. 
Do something like this in SignUp component:
`this.$store.dispatch('signUserUp', {/* payload */}).catch((error) => alert(error.message));`